pkg/api/apierrors/errors.go

```diff
@@ -2,60 +2,99 @@ package apierrors
 
 import (
 	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
 	"net/http"
+	"strings"
 
 	"github.com/gin-gonic/gin"
-	"github.com/numary/go-libs/sharedapi"
+	"github.com/numary/go-libs/sharedlogging"
 	"github.com/numary/ledger/pkg/ledger"
 	"github.com/numary/ledger/pkg/storage"
 	"github.com/pkg/errors"
 )
 
 const (
-	ErrInternal         = "INTERNAL"
-	ErrConflict         = "CONFLICT"
-	ErrInsufficientFund = "INSUFFICIENT_FUND"
-	ErrValidation       = "VALIDATION"
-	ErrContextCancelled = "CONTEXT_CANCELLED"
-	ErrStore            = "STORE"
-	ErrNotFound         = "NOT_FOUND"
+	ErrInternal                = "INTERNAL"
+	ErrConflict                = "CONFLICT"
+	ErrInsufficientFund        = "INSUFFICIENT_FUND"
+	ErrValidation              = "VALIDATION"
+	ErrContextCancelled        = "CONTEXT_CANCELLED"
+	ErrStore                   = "STORE"
+	ErrNotFound                = "NOT_FOUND"
+	ErrScriptCompilationFailed = "COMPILATION_FAILED"
+	ErrScriptNoScript          = "NO_SCRIPT"
+	ErrScriptMetadataOverride  = "METADATA_OVERRIDE"
 
 	errorCodeKey = "_errorCode"
 )
 
-func coreErrorToErrorCode(err error) (int, string) {
+func coreErrorToErrorCode(c *gin.Context, err error) (int, string, string) {
```

Comment:
The link is not a consideration of this function. This has to be handled at upper level.

Reply:
Ok, got it, it is for details.
Acceptable.